### ISSUE TITLE
Trigger message from "external"

### ISSUE THREAD

**Issue by ScuttleSE** [guidone/node-red-contrib-chatbot#15]
Maybe I'm an idiot, but I can't figure out how to trigger a message from the bot to the user without starting with an "In" object.

To take a very simple example;, I want to create a flow that sends a telegram-message when I get a HTTP-request on http://nodered.example.com/sendmessage

It would be excellent with an example like that in the documentation

**Comment by guidone** [comment]
You have to use a `Conversation node`, you have to specify the chatId, every conversation bot-user has a unique chatId, you can discover it with the `Debug node`

Yes a tutorial about that would be useful

**Comment by ScuttleSE** [comment]
Hrm... seems to need something more than Conversation.

This is the flow I just created: 
![tgflow1](https://cloud.githubusercontent.com/assets/4344284/18355984/50cc3e74-75eb-11e6-9626-d66a48dc466c.PNG)

I get the error "This node is not available for transport: null" and it seems to be attached to the Message-node

**Comment by guidone** [comment]
You're right, it doesn't get the right transport anywhere.
I'm fixing it

**Comment by guidone** [comment]
OK @ScuttleSE grab version 0.5.10, you can select the transport with a drop down in the conversation node configuration

**Comment by ScuttleSE** [comment]
Worked just fine.

**Comment by marcostrullato** [comment]
Hi, I still have the same issue. I'm using 0.6.24. 
Is it maybe a regression?

Thanks

**Comment by guidone** [comment]
It should work fine, there are tests for this

https://github.com/guidone/node-red-contrib-chatbot/blob/master/test/conversation-node.js

Is it possible that in your flow there's a function node that strips away this information?
Is it a flow that you can export here so I can look at it?

**Comment by marcostrullato** [comment]
Hi, I exported a test flow: it might be I'm wiring the nodes incorrectly.. let me know :-)

Thanks

[{"id":"3eee808b.b2679","type":"chatbot-telegram-send","z":"d7d606ad.3dd17","bot":"","track":false,"outputs":0,"x":1004.0079905192058,"y":469.714243888855,"wires":[]},{"id":"ff020ecc.28235","type":"function","z":"d7d606ad.3dd17","name":"alert via telegram","func":"msg.payload = 'Internal temperature reached X degrees!';\nreturn msg;","outputs":1,"noerr":0,"x":560.7143046061199,"y":379.99997901916504,"wires":[["a7fc7867.cc4928","838af521.57fb"]]},{"id":"838af521.57fb","type":"debug","z":"d7d606ad.3dd17","name":"","active":true,"console":"false","complete":"payload","x":803.0952402750652,"y":505.23806858062744,"wires":[]},{"id":"19b1e786.460328","type":"chatbot-telegram-receive","z":"d7d606ad.3dd17","bot":"","x":148.83332061767578,"y":295.6666564941406,"wires":[["f648f9c6.855dd8"]]},{"id":"f648f9c6.855dd8","type":"chatbot-conversation","z":"d7d606ad.3dd17","name":"conversation","chatId":"xxxxx","transport":"telegram","x":347.50000508626306,"y":333.3333088556925,"wires":[["ff020ecc.28235"]]},{"id":"a7fc7867.cc4928","type":"chatbot-message","z":"d7d606ad.3dd17","name":"","message":[""],"answer":false,"track":false,"parse_mode":"","x":809.1666475931804,"y":411.66664028167725,"wires":[["3eee808b.b2679"]]},{"id":"1e4700ac.2e9ba7","type":"inject","z":"d7d606ad.3dd17","name":"","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"x":163,"y":460.33331298828125,"wires":[["f648f9c6.855dd8"]]}]

**Comment by guidone** [comment]
Just remove the first empty text of the text node

<img width="501" alt="remove" src="https://cloud.githubusercontent.com/assets/502519/26342367/21859936-3f98-11e7-9cad-706dde71ff0c.png">

TL;DR

The `message node` uses the input payload only if the message field in the configuration panel is empty, the bug is that a list with just an empty element is considered not empty.

I've already fixed the bug in 0.6.25-beta-1, but you don't need to install this, just remove the empty element

**Comment by marcostrullato** [comment]
Great! it works!

Thank you
